### PR TITLE
Update Settings functions to use standard non deprecated Civi::settin…

### PR DESCRIPTION
**Pull request by seamuslee001**
…gs facade

ping @adixon @KarinG 

This is in response to the test failures shown up here https://test.civicrm.org/job/CiviCRM-Ext-Matrix/BKPROF=min,BUILDTYPE=git,CIVIVER=master,EXTKEY=com.iatspayments.civicrm,label=bknix-tmp/1582/

also as per the documentation here https://docs.civicrm.org/dev/en/latest/framework/setting/#setting-storage-and-retrieval

### CODE REVIEW

**Comment by KarinG** [comment]
Hey @seamuslee001 - thank you for this. Is this a safe edit for ESR 5.13.x?

**Comment by seamuslee001** [comment]
Hi @KarinG, AFAIK yes, of course doing an r-run will tell you :-)

**Comment by adixon** [comment]
Thanks @seamuslee001 . And here I thought I was keeping up ...

File: CRM/Iats/Upgrader.php [hunk 1]
```diff
@@ -13,8 +13,8 @@ public function getCurrentRevision() {
     try {
       $xmlfile = CRM_Core_Resources::singleton()->getPath('com.iatspayments.civicrm','info.xml');
       $myxml = simplexml_load_file($xmlfile);
-      $version = (string)$myxml->version;
-      CRM_Core_BAO_Setting::setItem($version, 'iATS Payments Extension', 'iats_extension_version');
+      $version = (string) $myxml->version;
+      Civi::settings()->set('iats_extension_version', $version);
```
Comment (KarinG) [inline CRM/Iats/Upgrader.php]: Is that a syntax fix @seamuslee001 ?

Comment (seamuslee001) [reply CRM/Iats/Upgrader.php]: Yes there is usually meant to be a space between the casting and variable AFAIK

File: CRM/Iats/Form/Settings.php [hunk 2]
```diff
@@ -139,7 +139,7 @@ public function postProcess() {
         unset($values[$key]);
       }
     }
-    CRM_Core_BAO_Setting::setItem($values, 'iATS Payments Extension', 'iats_settings');
+    Civi::settings()->set('iats_settings', $values);
     parent::postProcess();
```
Comment (KarinG) [inline CRM/Iats/Form/Settings.php]: I thought I saw a `get` here - did you just fix that?

**Comment by KarinG** [comment]
Patch applied cleanly - settings page loads correctly. Changing all values to other values and re-saving that correctly loads the new values. 

I though I saw a mistake -> a `get `that should be a `set`- but I don't see it anymore. I think @seamuslee001 fixed it. 

Over to you @adixon

File: CRM/Iats/Form/Settings.php (hunk 2 above)
Comment (seamuslee001) [reply CRM/Iats/Form/Settings.php]: Correct @KarinG :)

**Comment by adixon** [comment]
Okay, I'll review, do one more test on a live site, merge and then generate a new release.

**Comment by adixon** [comment]
New release 1.7.1 includes this PR. Thanks Karin and Seamus!

**Pull request merged**